test/Utils/AsyncQueue.ts

```diff
@@ -25,14 +25,33 @@ export class AsyncQueue<T> {
             return head;
         }
 
+        // keep reference to resolve function - if timeout finishes first, we don't want to wait for value!
+        let resolveToDelete: Function;
+
         const timeoutErr = getError(
             "TimeoutException", `Timeout exceeded waiting for element to arrive for ${timeout}.`);
         const timeoutPromise: Promise<T> = 
             new Promise((_, reject) => 
-                setTimeout(() => 
-                    reject(timeoutErr), timeout));
+                setTimeout(() => {
```

Comment:
Seems overly complex. Also this timeout is not cleared if it's completed before it. Not sure what's happening in `// we don't want to wait for value` section.

Reply:
there was a bug in async queue impl. 

In general when timeout passes we have to remove promise as we don't want to wait for its result any more. 

About clearing timeout - there is no need that, see:
```
 // when result comes first - don't mark item for deletion
        resultPromise.then(() => resolveToDelete = null);
```

beside that this is test code so I don't want care about clearning it.